cmd/cmd_test.go

```diff
@@ -1189,3 +1190,66 @@ func (c *FailCommandCustom) Info() *Info {
 func (c *FailCommandCustom) Run(context *Context, client *Client) error {
 	return c.err
 }
+
+func (s *S) TestVersionWithAPI(c *check.C) {
+	var stdout, stderr bytes.Buffer
+	mngr := NewManager("tsuru", "5.0", "", &stdout, &stderr, os.Stdin, nil)
+	var exiter recordingExiter
+	mngr.e = &exiter
+	command := version{manager: mngr}
+	context := Context{[]string{}, mngr.stdout, mngr.stderr, mngr.stdin}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		// Send response to be tested
+		rw.Write([]byte(`{"version":"1.7.4"}`))
+	}))
+
+	client := NewClient(&http.Client{}, &context, mngr)
+
+	os.Setenv("TSURU_TARGET", ts.URL)
+	defer os.Unsetenv("TSURU_TARGET")
+	err := command.Run(&context, client)
+	c.Assert(err, check.IsNil)
+	c.Assert(mngr.stdout.(*bytes.Buffer).String(),
+		check.Equals, "Client version: 5.0.\nServer version: 1.7.4\n")
+}
+
+func (s *S) TestVersionWithoutEnvVar(c *check.C) {
+	var stdout, stderr bytes.Buffer
+	mngr := NewManager("tsuru", "5.0", "", &stdout, &stderr, os.Stdin, nil)
+	var exiter recordingExiter
+	mngr.e = &exiter
+	command := version{manager: mngr}
+	context := Context{[]string{}, mngr.stdout, mngr.stderr, mngr.stdin}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		// Send response to be tested
+		rw.Write([]byte(`{"version":"1.7.4"}`))
+	}))
+
+	client := NewClient(ts.Client(), &context, mngr)
+	err := command.Run(&context, client)
+	c.Assert(err, check.NotNil)
```

Comment:
I think you forget the assertions about the client/server versions here... :smile: